transcom/mymove
# B 20740 and B 20296 INT

## [B-20740](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20740) and [B-20296](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20296)

> [!NOTE]  
> Currently failing QAE/CSR Playwright tests are due to another backlog that was merged into integration upstream of this one, and can be safely ignored for this backlog.

## Summary

This feature adds a SIT Cost Breakdown similar to what is seen for service items for HHG SIT. This was originally planned to be a UI-only feature last PI, but had to be delayed and expanded due to PPM SIT items being classified and handled in the code as moving expenses rather than HHG's service items equivalent.

In order for this feature to work properly, an existing SIT Costing Estimate function on the closeout page was leveraged, with additional data from the pricing rate engine added to the API response.

With the appropriate data being sent back to the UI, a similarly-appearing replica of the HHG SIT calculations pane was recreated to display when the closeout counselor is reviewing calculated SIT Costs.

Is there anything you would like reviewers to give additional scrutiny?

Please review the data handling in the backend and note if any glaring issues so they may be resolved. I did my best to emulate the existing PPM SIT calculation flows, but ppm sit calculations are tricky to follow and could use more granular unit testing than they currently have, so I may have dropped a piece of spaghetti or two.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the client app as a new customer. 
2. Create and submit a PPM.
3. Log in as a service counselor. 
4. Approve the PPM.
5. Log back in as the service member. 
6. Upload your PPM Documents, including at least one Storage type item. 
7. Submit the PPM again for closeout.
8. Log back in as a service counselor. 
9. Approve the first weight ticket, until the Storage item is reached. 
10. Click the "Show calculations" button under the Government SIT Cost amount.
11. Verify all data is populating correctly while altering the storage length, amount, and location.

### Frontend


### Backend



> [!TIP]
> A storybook component may be added if I can figure out how to mock a react hook that is imported into the component, as well as it's data. If someone is experienced with this, please let me know.
